File: web/plugins/is_number.js
import { WebPlugin } from "../engine/plugin.js";

export class IsNumberPlugin extends WebPlugin {
    constructor() {
        super("文字が数字か");
    }

    async execute(line, context) {
        const match = line.match(/^文字が数字か\s+「(.+)」$/);
        if (!match) return;

        context.runtime.write(
            /^[0-9]+(?:\.[0-9]+)?$/.test(match[1])
        );
    }
}

File: pc/plugins/is_number.py
import re
from core.plugin import Plugin


class IsNumberPlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(r'文字が数字か\s+「(.+)」', line)
        if not match:
            return None

        value = match.group(1)
        return f"print({value!r}.isdigit())"